components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/query/QueryManager.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.stream.processor.core.persistence.query;
+
+import org.wso2.carbon.config.ConfigurationException;
+import org.wso2.carbon.config.provider.ConfigProvider;
+import org.wso2.carbon.database.query.manager.QueryProvider;
+import org.wso2.carbon.database.query.manager.config.Queries;
+import org.wso2.carbon.database.query.manager.exception.QueryMappingNotAvailableException;
+import org.wso2.carbon.stream.processor.core.persistence.beans.PersistenceStoreConfigs;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.CustomClassLoaderConstructor;
+import org.yaml.snakeyaml.introspector.BeanAccess;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Holds the database queries.
+ */
+public class QueryManager {
+
+    private Map<String, String> queries;
+    private ConfigProvider deploymentConfigProvider;
+
+    public QueryManager(String databaseType, String databaseVersion, ConfigProvider configProvider) throws
+            QueryMappingNotAvailableException, ConfigurationException, IOException {
+        deploymentConfigProvider = configProvider;
+        queries = readConfigs(databaseType, databaseVersion);
+    }
+
+    private Map<String, String> readConfigs(String databaseType, String databaseVersion) throws ConfigurationException,
+            QueryMappingNotAvailableException, IOException {
+        try {
+            PersistenceStoreConfigs deploymentConfigurations = deploymentConfigProvider
+                    .getConfigurationObject(PersistenceStoreConfigs.class);
+            List<Queries> deploymentQueries = deploymentConfigurations.getQueries();
+            List<Queries> componentQueries;
+            URL url = this.getClass().getClassLoader().getResource("queries.yaml");
+            if (url != null) {
+                PersistenceStoreConfigs componentConfigurations = readYamlContent(url.openStream());
+                componentQueries = componentConfigurations.getQueries();
+            } else {
+                throw new RuntimeException("Unable to load queries.yaml file.");
```

Comment:
Why are we throwing a runtime exception here? Isn't it a IOException or a ConfigurationException.

Reply:
Fixed in commit #cda7601